transform/operations_test.go

```diff
@@ -375,21 +386,122 @@ func TestTimeParse(t *testing.T) {
 	runOpTests(t, func() transformOperation { return &timeParse{} }, tests)
 }
 
+func TestCurrentTime(t *testing.T) {
+	tests := []opTests{
+		{
+			description: "Simple working case",
+			args:        map[string]string{"format": "RFC3339"},
+			want:        time.Now().Format(time.RFC3339),
+		},
+		{
+			description: "Missing arg",
+			args:        map[string]string{},
+			wantInitErr: true,
+		},
+		{
+			description: "Too many args",
+			args:        map[string]string{"format": "RFC3339", "cookies": "failure"},
+			wantInitErr: true,
+		},
+		{
+			description: "Format not predefined",
+			args:        map[string]string{"format": "Mon Jan 2 15:04:05 MST 2006"},
+			want:        time.Now().Format("Mon Jan 2 15:04:05 MST 2006"),
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.description, func(t *testing.T) {
+			op, err := runOpTestInit(func() transformOperation { return &currentTime{} }, test)
+			if err != nil {
+				return
+			}
+			got, err := runOpTestTransform(op, test)
+
+			if err != nil {
+				t.Fatal(err)
+				return
+			}
+
+			if test.want == nil {
```

Comment:
why is this case needed?

Reply:
Because if this isn't there, my test (and only my test) tries to do the transform, even when the init is designed to fail.

Reply:
For more detail:
Other tests call 
```
// A common test runner for all the operations tests
func runOpTests(t *testing.T, opType func() transformOperation, tests []opTests) {
	for _, test := range tests {
		t.Run(test.description, func(t *testing.T) {
			runOpTest(t, opType, test)
		})
	}
}

func runOpTest(t *testing.T, opType func() transformOperation, test opTests) {
	op, err := runOpTestInit(opType, test)
	if err != nil {
		t.Fatal(err)
	}
	if test.wantInitErr {
		return
	}
	_, err = runOpTestTransform(op, test)
	if err != nil {
		t.Fatal(err)
	}
}
```

Which stops processing if init fails. But my test calls them individually.

Reply:
Right, but if you fatal out when the error returns on runOpTestInit then it won't get to that point right?